core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java

```diff
@@ -659,6 +650,33 @@ public String send(MessageProducer messageProducer, javax.jms.Message message, b
 		}
 	}
 
+	protected void logMessageDetails(javax.jms.Message message, MessageProducer messageProducer) throws JMSException {
+		if (log.isInfoEnabled() && !log.isDebugEnabled()) {
+				log.info("[" + getName()
+						+ (messageProducer != null ? " ] message destination: [" + messageProducer.getDestination() + "] " : "")
+						+ "] msgID: [" + message.getJMSMessageID()
+						+ "] correlationID: [" + message.getJMSCorrelationID()
+						+ "] using deliveryMode: [" + message.getJMSDeliveryMode() + "] "
+						+ (message.getJMSReplyTo() != null ? "replyTo: [" + message.getJMSReplyTo() + "]" : ""));
+				return;
+		}
+		if (!log.isDebugEnabled()) {
+			return;
+		}
+
```

Comment:
I think that I would find it more logical to not have 2x checking `!log.isDebugEnabled()` and here an early return, but instead structure the conditions like
```
if (log.isDebugEnabled()) {
    ...
} else if (log.isInfoEnabled()) {
   ...
}
```

A matter of personal style preference perhaps so not a big thing.

Reply:
I've changed it to: 
```
if (!log.isDebugEnabled()) {
   ....
   return;
}

log.debug(....).
```
assuming that info-logging is turned on mostly.